pkg/operator/controllers/checker/auth.go

```diff
@@ -0,0 +1,66 @@
+package checker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	jwt "github.com/form3tech-oss/jwt-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclaim"
+	"github.com/Azure/ARO-RP/pkg/util/refreshable"
+)
+
+type credentials struct {
+	clientID     []byte
+	clientSecret []byte
+	tenantID     []byte
+}
+
+func newAuthorizer(token *adal.ServicePrincipalToken) (refreshable.Authorizer, error) {
```

Comment:
1. Dynamic currently takes an authorizer, so it doesn't have the creds in it's struct which it would need.  I can refactor dynamic to take creds and create the authorizer itself and export a function from it's interface if we think that's a better solution.
2. I see the regression from operator's standpoint.  Fixed the function and call temporarily.